src/main/java/com/example/demo/service/ServiceFactory.java

```diff
@@ -0,0 +1,8 @@
+package com.example.demo.service;
+
+public class ServiceFactory {
```

Comment:
this implementation is wrong
1. should be an enum with two types: `enum ServiceType`
2. each service (for db and for redis) should have this property: `ServiceType getType()`
3. factory should autowire list of services and provide some by type: `public AutoService getService(ServiceType type) {`
4. please avoid to use static methods